### DISCUSSION TITLE
Do I have to deploy a model before doing inference?

### DISCUSSION THREAD

**Discussion by furtheraway** [microsoft/InnerEye-DeepLearning#670] [Q&A] [answered]
Hi all,

I have trained a model successfully in Azure machine leaning workspace, and now I am working on how to make inference from that model. I followed the following link: https://github.com/microsoft/InnerEye-Inference

But I don't know where to get the following two parameters: 

> export CUSTOMCONNSTR_AZUREML_SERVICE_PRINCIPAL_SECRET=
> export CUSTOMCONNSTR_API_AUTH_SECRET=

I guess I need to deploy my model in Azure first. Is there any guidence on how to do that? There seems a lot options to proceed:

![image](https://user-images.githubusercontent.com/7104230/155637554-748ab5c7-4347-4ab8-b79b-867e25889e70.png)

**Comment by ant0nsc** [comment, marked as answer]
There is no need to deploy the model manually. If you have run training successfully in AzureML, the job overview page ("Details") will have an entry for "Registered model" - click on that link. 
About the two environment variables: 
* `CUSTOMCONNSTR_AZUREML_SERVICE_PRINCIPAL_SECRET` is the password for an Azure Service Principal (think of that as a machine account) that can talk to your AzureML workspace. You need to create this service principal first, and give it access to your AzureML workspace.
* `CUSTOMCONNSTR_API_AUTH_SECRET` is a random string that you can choose, you can use a GUID for example. This is the magic secret that you put into the API call to authenticate yourself to the inference app.

**Reply by furtheraway** [reply]
Thank you @ant0nsc !

Looks like the "Models" is located on the left pane now.  
Thank you for explaining the two configuration parameters!

![image](https://user-images.githubusercontent.com/7104230/156492884-a0acb43b-314f-49f8-9294-fc79e0876377.png)
